portway/portway
# Notifications & Errors

* Adds new Notifications store/actions/components
* Adds error handling for `fetch` on projects and documents
* Adds in ErrorBoundary to catch any React errors so the user doesn't get a blank screen
  * Added a commented function in there for when we ever want to log app errors somewhere
* Getting eyes on this before going all the way with all the things (post, put, remove, etc)

To test:

* Log in as bobx@bong.com, and you'll see he has no permissions to [here](http://localhost:3000/d/project/110000) and [here](http://localhost:3000/d/projects)
* Break a component by not returning anything (remove a `null` from a `return null` in the UserMenu), and refresh the page

Questions:
* It'd be cool to have a unique message per resource, notice the following when a user doesn't have the project or documents permissions. How can we provide messages like "You don;'t have access to these documents" and "Sorry, this project is not available to you" sort of things?

<img width="1392" alt="Screen Shot 2019-05-08 at 2 54 13 PM" src="https://user-images.githubusercontent.com/2809/57410983-266f1c00-71a1-11e9-9b85-e5697dc4129b.png">
